File: src/main/java/it/discovery/loader/BootstrapLoader.java
package it.discovery.loader;

import java.time.LocalDateTime;
import java.util.List;

import it.discovery.model.Book;
import it.discovery.model.BookFactory;
import it.discovery.model.BookState;
import it.discovery.model.Genre;
import it.discovery.service.MainBookService;

public class BootstrapLoader {

	public static void main(String[] args) {
		MainBookService service = new MainBookService();
		Book book = BookFactory.createBook(BookState.PRESENT);
		book.setName("Introduction into Kotlin");
		book.setPages(100);
		book.setYear(2021);

		Genre genre = new Genre("IT", LocalDateTime.now());
		book.setGenre(genre);

		service.saveBook(book);

		List<Book> books = service.findBooks();
		System.out.println(books);

		Book book1 = service.findBookById(book.getId()).orElseThrow();
		System.out.println(book1);
	}

}

File: src/main/kotlin/it/discovery/loader/BootstrapLoader.kt
package it.discovery.loader

import it.discovery.model.BookFactory
import it.discovery.model.BookState
import it.discovery.model.Genre
import it.discovery.service.MainBookService
import java.time.LocalDateTime

fun main(args: Array<String>) {
    val service = MainBookService()
    val book = BookFactory.createBook(BookState.PRESENT)
    book.setName("Introduction into Kotlin")
    book.setPages(100)
    book.setYear(2021)

    val genre = Genre("IT", LocalDateTime.now())
    book.setGenre(genre)

    service.saveBook(book)

    val books = service.findBooks()
    println(books)

    val book1 = service.findBookById(book.getId()).orElseThrow()
    println(book1)
}